tests/validation/cattlevalidationtest/core/common_fixtures.py

```diff
@@ -361,3 +361,21 @@ def wait_for_condition(client, resource, check_function, fail_handler=None,
         resource = client.reload(resource)
 
     return resource
+
+
+@pytest.fixture
+def wait_until_expected_state(client, resource, expected_state,
```

Comment:
@son-rancher Just recently added a wait_for_condition function that does something very similar, but more generically. You could probably use that to avoid some code duplication.